File: Algorithm/baekjoon/java/BJ10610.java
package baekjoon.java;

import java.lang.reflect.*;
import java.util.*;

public class BJ10610 {
    public static String max30(String input){
        char[] tokens = input.toCharArray();
        Arrays.sort(tokens);

        int sum = 0;
        for(int i = 0; i < input.length(); ++i){
            int value = tokens[i] - '0';
            sum += value;
        }

        if(tokens[0] == '0' && sum % 3 == 0) {
            for (int i = 0; i < tokens.length / 2; ++i) {
                char temp = tokens[i];
                tokens[i] = tokens[tokens.length - 1 - i];
                tokens[tokens.length - 1 - i] = temp;
            }
            String output = new String(tokens);
            return output;
        }
        else
            return "-1";
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        String input = scanner.nextLine();
        String output = max30(input);
        System.out.println(output);
    }
}

File: Algorithm/baekjoon/swift/BJ10610.swift
//10610
var input = readLine()

if let input = input{
    let tokens = input.map({(value:Character) in return String(value)})
    //let tokens = input.map({(value:Character) in String(value)})
    //let tokens = input.map({(value) -> String in String(value)})
    //let tokens = input.map({value in String(value)})
    //let tokens = input.map({String($0)})
    //let tokens = input.map{String($0)}
    var sortedTokens = tokens.sorted()
    let length = tokens.count
    var sum = 0
    
    for i in 0..<length{
        sum += Int(sortedTokens[i])!
    }
    
    if sortedTokens[0] == "0" && sum % 3 == 0{
        var output : String = ""
        for i in 0..<length/2{
            let temp = sortedTokens[length - 1 - i]
            sortedTokens[length - 1 - i] = sortedTokens[i]
            sortedTokens[i] = temp
        }
        for i in 0..<length{
            output.append(sortedTokens[i])
        }
        
        print(output)
    }else{
        print(-1)
    }
}
